aspnetcore/fundamentals/w3c-logger/index.md

```diff
@@ -0,0 +1,116 @@
+uid: fundamentals/w3c-logger/index
+---
+
+# W3C Logger in ASP.NET Core
+
+W3CLogger is a middleware that writes log files, in the W3C standard format (described [here](https://www.w3.org/TR/WD-logfile.html)), containing information about HTTP requests and HTTP responses. W3C logger provides logs of:
+
+* HTTP request information
+* Common properties
+* Headers
+* HTTP response information
+* Metadata about the request/response pair (date/time started, time taken)
+
+W3C Logging is valuable in several scenarios to:
+
+* Record information about incoming requests and responses.
+* Filter which parts of the request and response are logged.
+* Filtering which headers to log.
+
+W3CLogger ***can reduce the performance of an app***. Consider the performance impact when selecting fields to log. Test the performance impact of the selected logging properties.
+
+> [!WARNING]
+> W3CLogger can potentially log personally identifiable information (PII). Consider the risk and avoid logging sensitive information. By default, fields that could contain PII are not logged.
+
+## Enabling W3C Logging
+
+W3C Logging is enabled with `UseW3CLogging`, which adds the W3CLogger middleware.
+
+[!code-csharp[](samples/6.x/Startup.cs?name=snippet&highlight=3)]
+
+By default, W3CLogger logs common properties such as path, status-code, date, time, and protocol. All information about a single request/response pair is written to the same line.
+
+```
+#Version: 1.0
+#Start-Date: 2021-09-29 22:18:28
+#Fields: date time c-ip s-computername s-ip s-port cs-method cs-uri-stem cs-uri-query sc-status time-taken cs-version cs-host cs(User-Agent) cs(Referer)
+2021-09-29 22:18:28 ::1 DESKTOP-LH3TLTA ::1 5000 GET / - 200 59.9171 HTTP/1.1 localhost:5000 Mozilla/5.0+(Windows+NT+10.0;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/93.0.4577.82+Safari/537.36 -
+2021-09-29 22:18:28 ::1 DESKTOP-LH3TLTA ::1 5000 GET / - 200 0.1802 HTTP/1.1 localhost:5000 Mozilla/5.0+(Windows+NT+10.0;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/93.0.4577.82+Safari/537.36 -
+2021-09-29 22:18:30 ::1 DESKTOP-LH3TLTA ::1 5000 GET / - 200 0.0966 HTTP/1.1 localhost:5000 Mozilla/5.0+(Windows+NT+10.0;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/93.0.4577.82+Safari/537.36 -
+```
+
+## W3C Logger options
+
+To configure the W3CLogger middleware, call `AddW3CLogging` in `ConfigureServices`.
+
+[!code-csharp[](samples/6.x/Startup.cs?name=configureservices)]
+
+### `LoggingFields`
+
+`W3CLoggerOptions.LoggingFields` is an enum flag that configures specific parts of the request and response to log, as well as other information about the connection. `LoggingFields` defaults to include all possible fields except `UserName` and `Cookie`. 
+
+[!code-csharp[](samples/6.x/Startup.cs?name=configureservices&highlight=6)]
+
+| Flag | Description | Value |
```

Comment:
Should this table be here or should we just link to the API doc page for W3CLoggingFields when that's ready?

Reply:
I copied what we do in the HttpLogging doc - https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-logging/?view=aspnetcore-6.0#loggingfields. @Rick-Anderson should both of these change to reference the API doc page after 6.0 ships?

Reply:
Link to the API docs, we don't want to have to maintain both.